app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -442,8 +446,14 @@ private void updateLoyaltyCardList(boolean updateCount) {
         if (mCurrentActionMode != null) {
             mCurrentActionMode.finish();
         }
-    }
 
+        updateWidget(mAdapter.mContext);
```

Comment:
It seems kind of incorrect to update the widget every time the list of cards changes, as it should be more dependent on database. But for version 1 I think this is fine, it would be a rather small optimization :)